### PR TITLE
Upgrade to latest Katamari/Sugar and fixed a few issues reported

### PR DESCRIPTION
**Pull request by lnewson**

### CODE REVIEW

File: src/main/ts/Part2Ex4FP.ts [hunk 1]
```diff
@@ -141,7 +138,7 @@ You can find this as Fun.constant in katamari.
 One way of writing it is below:
  */
 
-const constant = <A, B> (a: A) => (b: B) => a;
+const constant = <A> (a: A) => (...args: any[]): A => a;
```
Comment (lnewson) [inline src/main/ts/Part2Ex4FP.ts:141]: The required `b` argument caused typescript to complain when used in `fold` below.

File: src/main/ts/Part2Ex3Optional.ts [hunk 2]
```diff
@@ -117,10 +114,10 @@ export const unsafeStuff = (e: Optional<string>): void => {
 
 A common way to handle an Optional value is to provide a default value if in the case of none.
 
-You can do this with fold, but getOrElse is a shortcut.
+You can do this with fold, but getOr is a shortcut.
```
Comment (lnewson) [inline src/main/ts/Part2Ex3Optional.ts:117]: `getOrElse` doesn't exist on `Optional` so this caused some confusion.

**Review by ashfordneil — APPROVED** [review]

**Review by ltrouton — APPROVED** [review]

File: src/main/ts/Part2Ex4FP.ts (hunk 1 above)
Comment (tinydylan) [reply src/main/ts/Part2Ex4FP.ts:141]: Would `unknown[]` work here instead?

**Review by tinydylan — APPROVED** [review]

File: src/main/ts/Part2Ex4FP.ts (hunk 1 above)
Comment (lnewson) [reply src/main/ts/Part2Ex4FP.ts]: Yeah it would, I've switched it over.

**Pull request merged**